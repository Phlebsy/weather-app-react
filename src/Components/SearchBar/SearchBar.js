import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { retrieveWeatherData } from '../../context/actions';

// SearchBar component handles the input for our location
// Logic includes an onClick handler to call the action to retrieve weather data
export default class SearchBar extends Component {
  state = {
    searchInput: 'New York, NY'
  };

  onChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  getWeatherData = async () => {
    const { dispatch } = this.props.store;
    retrieveWeatherData(dispatch, this.state.searchInput);
  };

  render() {
    return (
      <div className="searchbar container d-flex justify-content-center my-2">
        <label htmlFor="searchInput" className="pr-2 font-weight-bold">
          City:{' '}
        </label>
        <input
          name="searchInput"
          value={this.state.searchInput}
          type="text"
          onChange={this.onChange}
        />
        <button className="btn-primary" onClick={this.getWeatherData}>
          Go
        </button>
      </div>
    );
  }
}
