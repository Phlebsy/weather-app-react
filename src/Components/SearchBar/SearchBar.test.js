import React from 'react';
import ReactDOM from 'react-dom';
import SearchBar from './SearchBar';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('<SearchBar />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SearchBar />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches the snapshot', () => {
    const tree = shallow(<SearchBar />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
