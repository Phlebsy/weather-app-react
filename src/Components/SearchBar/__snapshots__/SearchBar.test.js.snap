// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> matches the snapshot 1`] = `
<div
  className="searchbar container d-flex justify-content-center my-2"
>
  <label
    className="pr-2 font-weight-bold"
    htmlFor="searchInput"
  >
    City:
     
  </label>
  <input
    name="searchInput"
    onChange={[Function]}
    type="text"
    value="New York, NY"
  />
  <button
    className="btn-primary"
    onClick={[Function]}
  >
    Go
  </button>
</div>
`;
